plugins/BEdita/Core/src/Mailer/UserMailer.php

```diff
@@ -38,4 +41,50 @@ public function welcome($options)
             ->setTo($options['to'])
             ->setSubject('Welcome!');
     }
+
+    /**
+     * Signup message.
+     *
+     * It requires `$options['params']` with:
+     * - `userId` the user id to send signup email
+     * - `activationUrl` the activation url to follow
+     *
+     * @param array $options Email options
+     * @return void
+     * @throws \LogicException When missing some required parameter
+     */
+    public function signup(array $options)
+    {
+        if (empty($options['params']['userId'])) {
+            throw new \LogicException(__d('bedita', 'Parameter "{0}" missing', ['params.userId']));
+        }
+
+        if (empty($options['params']['activationUrl'])) {
+            throw new \LogicException(__d('bedita', 'Parameter "{0}" missing', ['params.activationUrl']));
+        }
+
+        $users = TableRegistry::get('Users');
+        $action = new GetObjectAction(['table' => $users]);
+        $user = $action(['primaryKey' => $options['params']['userId']]);
+
+        if (empty($user->email)) {
+            throw new \LogicException(__d('bedita', 'User email missing'));
+        }
+
+        $currentApplication = CurrentApplication::getApplication();
+        $appName = ($currentApplication !== null) ? $currentApplication->name : 'BEdita';
+        $subject = __d('bedita', '{0} registration', [$appName]);
+
+        $this->set([
+            'user' => $user,
+            'activationUrl' => $options['params']['activationUrl'],
+            'appName' => $appName
+        ]);
+
+        $this->setTemplate('BEdita/Core.signup')
```

Comment:
Here it would probably be better to `return` the whole chain, so that a mail preview can be set up with DebugKit (see http://yourlocalbeditaurl.localhost/debug_kit/mail_preview).